Zend/zend_compile.c

```diff
@@ -7578,8 +7600,17 @@ static void zend_compile_attributes(
 				}
 			}
 
-			if (config->validator != NULL) {
-				config->validator(attr, target, CG(active_class_entry));
+			// Validators are not run if the target is already invalid
+			if (run_validator && config->validator != NULL) {
+				zend_string *error = config->validator(attr, target, CG(active_class_entry));
+				if (error != NULL) {
+					if (delayed_target_validation == false) {
```

Comment:
You are comparing a pointer to false:
```suggestion
					if (delayed_target_validation == NULL) {
```

Reply:
Yes, `false` is just `0` with C11:
```
#define false	0
```
And `0` is a NULL pointer literal. `NULL` itself is typically defined as `#define NULL (void *)0`, though.